server/src/main/java/org/opensearch/index/translog/transfer/FileSnapshot.java

```diff
@@ -108,18 +110,51 @@ public static class TransferFileSnapshot extends FileSnapshot {
 
         private final long primaryTerm;
         private Long checksum;
+        @Nullable
+        private long generation;
+        @Nullable
+        private long minTranslogGeneration;
+        @Nullable
+        private Path ckpFilePath;
+        @Nullable
+        private Long ckpFileChecksum;
+        @Nullable
+        private Map<String, String> metadata;
+
+        public TransferFileSnapshot(Path path, long primaryTerm, long generation, Long checksum) throws IOException {
+            super(path);
+            this.primaryTerm = primaryTerm;
+            this.checksum = checksum;
+            this.generation = generation;
+        }
 
-        public TransferFileSnapshot(Path path, long primaryTerm, Long checksum) throws IOException {
+        public TransferFileSnapshot(
+            long primaryTerm,
+            long generation,
+            long minTranslogGeneration,
+            Path path,
+            Long checksum,
+            Path ckpFilePath,
+            Long ckpFileChecksum
+        ) throws IOException {
```

Comment:
This comment is for the entire class. With the change it looks more messier. We can just add metadata capability to the parent class FileSnapshot as this can be nature of any file in general. We can, otherwise, leave the rest of the classes same as earlier. In the `TranslogTransferManager` class we can have a interface that accepts a new class lets say `TranslogCheckpointSnapshot` which will have methods to return the `TranslogFileSnapshot`, `CheckpointFileSnapshot` and `TranslogFileSnapshot` with metadata. We can have an Factory that supplies the implementation based on the `isBlobMetadataSupported`. This will keep the classes sleek and understandable.